Add lazily recipe

justbur opened the pull request:
### Brief summary of what the package does

lazily provides a macro to execute forms in a config file while catching any
void-variable or void-function errors. The bad forms are stored and retried
after every new load using after-load-functions.

### Direct link to the package repository

https://github.com/justbur/emacs-lazily

### Your association with the package

Author/maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

purcell commented:
Interesting. Seems to me that this is likely to result in forms being executed over and over again in quite unpredictable ways at startup, thus slowing everything down. And any typos which result in legitimate `void-variable` errors etc. will never be surfaced to the user. I'm quite wary of promoting this approach: once on MELPA this package will be found by Emacs newbies and they won't realise the pitfalls.

justbur commented:
> Interesting. Seems to me that this is likely to result in forms being executed over and over again in quite unpredictable ways at startup, thus slowing everything down. 

I've set it up to check if for example there is a void-variable error that the variable is defined before the form is tried again to avoid this issue. I've also added two variants, `lazily-do` and `lazily-do-all`. One stops as soon as any error is found and stores all remaining forms. The `-all` version keeps going. 

> And any typos which result in legitimate void-variable errors etc. will never be surfaced to the user. I'm quite wary of promoting this approach: once on MELPA this package will be found by Emacs newbies and they won't realise the pitfalls.

Fair enough. This is an experiment as far as I'm concerned. Agreed that typos are one of the main issues. I added a function that produces a report of everything found. If you're not comfortable with it on MELPA, I have no problem with that.
